projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,312 @@
+import { Component, Element, Event, EventEmitter, Method, Prop, State, Watch } from '@stencil/core';
+import Fragment from 'stencil-fragment';
+import { DotFieldStatus, DotFieldStatusEvent, DotFieldValueEvent } from '../../models';
+import {
+    checkProp,
+    getClassNames,
+    getDotOptionsFromFieldValue,
+    getErrorClass,
+    getHintId,
+    getId,
+    getOriginalStatus,
+    getTagError,
+    getTagHint,
+    updateStatus
+} from '../../utils';
+
+const URL_REGEX = new RegExp(
+    /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/
+);
+
+/**
+ * Represent a dotcms binary file control.
+ *
+ * @export
+ * @class DotBinaryFile
+ */
+@Component({
+    tag: 'dot-binary-file',
+    styleUrl: 'dot-binary-file.scss'
+})
+export class DotBinaryFileComponent {
+    @Element() el: HTMLElement;
+
+    /** Value specifies the value of the <input> element */
+    @Prop({ mutable: true })
+    value = null;
+
+    /** Name that will be used as ID */
+    @Prop() name = '';
+
+    /** (optional) Text to be rendered next to input field */
+    @Prop({ reflectToAttr: true })
+    label = '';
+
+    /** (optional) Placeholder specifies a short hint that describes the expected value of the input field */
+    @Prop({ reflectToAttr: true })
+    placeholder = 'Attach files by dragging & dropping, selecting or pasting them.';
+
+    /** (optional) Hint text that suggest a clue of the field */
+    @Prop({ reflectToAttr: true })
+    hint = '';
+
+    /** (optional) Determine if it is mandatory */
+    @Prop({ reflectToAttr: true })
+    required = false;
+
+    /** (optional) Text that be shown when required is set and condition not met */
+    @Prop() requiredMessage = 'This field is required';
+
+    /** (optional) Text that be shown when the Regular Expression condition not met */
+    @Prop() validationMessage = "The field doesn't comply with the specified format";
+
+    @Prop() URLValidationMessage = 'The specified URL is not valid';
+
+    /** (optional) Disables field's interaction */
+    @Prop({ reflectToAttr: true })
+    disabled = false;
+
+    /** (optional) Describes a type of file that may be selected by the user, separated by comma  eg: .pdf,.jpg  */
+    @Prop({ reflectToAttr: true })
+    accept = '';
+
+    /** (optional) Text that be shown in the browse file button */
+    @Prop({ reflectToAttr: true })
+    buttonLabel = 'Browse';
+
+    @State() status: DotFieldStatus;
+    @State() errorElement: JSX.Element = null;
```

Comment:
Shouldn't be the message the state and not the element?

Reply:
changed to follow standard.